### PR TITLE
build: adds product dashboard to vaults page

### PR DESCRIPTION
**Pull request by matthew-garrett**
- Adds InfiniteScrollDataTable to MUI
- Adds Analytics route
- Adds react-query from tan stack

### CODE REVIEW

File: apps/web/src/containers/AnalyticsView/hooks/use-analytics-table.tsx [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+  MultiValueCell,
+  TxnHashCell,
+  DateTimeCell,
+} from '@notional-finance/mui';
+import { FormattedMessage } from 'react-intl';
+
+export const useAnalyticsTable = () => {
+  const columns = useMemo<Array<any>>(
+    () => [
+      {
```
Comment (jeffywu) [inline apps/web/src/containers/AnalyticsView/hooks/use-analytics-table.tsx]: Can you add the account address as a column in the analytics table?

File: apps/web/src/containers/AnalyticsView/hooks/use-analytics-table.tsx [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+      },
+      {
+        header: (
+          <FormattedMessage defaultMessage="Time" description={'Time header'} />
+        ),
+        cell: DateTimeCell,
+        accessorKey: 'time',
+        textAlign: 'right',
+      },
+      {
```
Comment (jeffywu) [inline apps/web/src/containers/AnalyticsView/hooks/use-analytics-table.tsx]: Can you add the account as a column here? Make it the second to last column,

File: apps/web/src/containers/AnalyticsView/hooks/use-formatted-data.tsx [hunk 3]
```diff
@@ -0,0 +1,68 @@
+import { Network, getEtherscanTransactionLink } from '@notional-finance/util';
+import { ReceivedIcon, SentIcon } from '@notional-finance/icons';
+import { formatTokenAmount, formatTokenType } from '@notional-finance/helpers';
+import { useMemo } from 'react';
+import { useTheme } from '@mui/material';
+
+export const useFormattedData = (data: any[], network?: Network) => {
```
Comment (jeffywu) [inline apps/web/src/containers/AnalyticsView/hooks/use-formatted-data.tsx]: Can this be reused in `use-txn-history-data`? Prefer not to have duplicate logic

File: apps/web/src/containers/App/App.tsx [hunk 4]
```diff
@@ -187,6 +188,11 @@ const AllRoutes = () => {
             component={Markets}
             routeType="Markets"
           />
+          <AppLayoutRoute
+            path="/analytics"
```
Comment (jeffywu) [inline apps/web/src/containers/App/App.tsx]: Can you route this to `/analytics/transactions`? In the future we will have more views under this.

Can you also change the routeType for this to `Analytics`

File: apps/web/src/containers/App/App.tsx [hunk 5]
```diff
@@ -187,6 +188,11 @@ const AllRoutes = () => {
             component={Markets}
             routeType="Markets"
```
Comment (jeffywu) [inline apps/web/src/containers/App/App.tsx]: Can you move `/markets` into `/analytics/markets` and then change the route type on both of them to `Analytics`?

File: apps/web/src/containers/AnalyticsView/index.tsx [hunk 6]
```diff
@@ -0,0 +1,187 @@
...
+import { useAnalyticsTable } from './hooks/use-analytics-table';
+import { Registry } from '@notional-finance/core-entities';
+import { Network, getEtherscanTransactionLink } from '@notional-finance/util';
+import { ReceivedIcon, SentIcon, StackIcon } from '@notional-finance/icons';
+import { formatTokenAmount, formatTokenType } from '@notional-finance/helpers';
+import { useNetworkToggle } from './hooks/use-network-toggle';
+import { useTheme, Box, styled } from '@mui/material';
+import { colors } from '@notional-finance/styles';
+
+export const AnalyticsView = () => {
```
Comment (jeffywu) [inline apps/web/src/containers/AnalyticsView/index.tsx]: Can you rename this file to `NetworkTransactionsView`? I will want to put additional views in here in the future.

Can you rename the parent folder from `AnalyticsView` => `AnalyticsViews`?

File: packages/common/mui/src/lib/contest-table/contest-table.tsx [hunk 7]
```diff
@@ -142,4 +142,6 @@ export const ContestTable = ({
   );
 };
 
+// export default ContestTable;
```
Comment (jeffywu) [inline packages/common/mui/src/lib/contest-table/contest-table.tsx]: Remove the commented code

File: packages/core-entities/src/client/analytics-registry-client.ts [hunk 8]
```diff
@@ -445,7 +445,10 @@ export class AnalyticsRegistryClient extends ClientRegistry<unknown> {
       (r): Record<string, AccountHistory[]> => {
         return {
           [network]: r.transactions
-            .map((t) => parseTransaction(t as Transaction, network))
+            .map((t) => {
+              const test = parseTransaction(t as Transaction, network)
+              return test
```
Comment (jeffywu) [inline packages/core-entities/src/client/analytics-registry-client.ts]: Revert this change

File: packages/features/portfolio/src/containers/portfolio-txn-history/hooks/use-txn-history-data.tsx [hunk 9]
```diff
@@ -14,7 +14,6 @@ export const useTxnHistoryData = (txnHistoryCategory: number) => {
   const theme = useTheme();
   let assetOrVaultData: SelectedOptions[] = [];
   let currencyData: SelectedOptions[] = [];
-
   const network = useSelectedNetwork();
```
Comment (jeffywu) [inline packages/features/portfolio/src/containers/portfolio-txn-history/hooks/use-txn-history-data.tsx:15]: This hook should re-use the `use-formatted-data` from analytics view, I think we should also re-use the column configuration as well so the things are more or less the same.

In the per account view, however, the only change would be to remove the account column.

File: packages/shared/web/src/footer/footer.tsx [hunk 10]
```diff
@@ -168,6 +168,16 @@ export const Footer = () => {
               description="footer link"
             />
           </Label>
+          <HeadingSubtitle
+            contrast
+            to="/analytics"
+            marginBottom={theme.spacing(4)}
+          >
+            <FormattedMessage
+              defaultMessage="Analytics"
```
Comment (jeffywu) [inline packages/shared/web/src/footer/footer.tsx]: Rename this to "View Network Transactions"

File: apps/web/src/containers/AnalyticsView/hooks/use-network-toggle.tsx [hunk 11]
```diff
@@ -0,0 +1,63 @@
+import { useTheme } from '@mui/material';
+import { TokenIcon } from '@notional-finance/icons';
+import { useState } from 'react';
+import { Body } from '@notional-finance/mui';
+import { FormattedMessage } from 'react-intl';
+
+export const useNetworkToggle = () => {
```
Comment (jeffywu) [inline apps/web/src/containers/AnalyticsView/hooks/use-network-toggle.tsx]: Can we create a re-usable compoent for this? Isn't is used in multiple places?

Furthermore, instead of hardcoding the network options like this we should loop over SUPPORTED_NETWORKS and generate the options. That will be more scalable as we add more networks and then we dont have to come in and change all this stuff.

File: packages/common/mui/src/lib/infinite-scroll-data-table/infinite-scroll-data-table.tsx [hunk 12]
```diff
@@ -0,0 +1,161 @@
...
+    data: flatData,
+    columns: columns,
+    getCoreRowModel: getCoreRowModel(),
+  });
+  const { rows } = table.getRowModel();
+  const headerGroups = table.getHeaderGroups();
+
+  const rowVirtualizer = useVirtualizer({
+    count: rows.length,
+    estimateSize: () => 50, //estimate row height for accurate scrollbar dragging
```
Comment (jeffywu) [inline packages/common/mui/src/lib/infinite-scroll-data-table/infinite-scroll-data-table.tsx]: Is this in px or row count?

Comment (matthew-garrett) [reply packages/common/mui/src/lib/infinite-scroll-data-table/infinite-scroll-data-table.tsx]: It is px height. I increased it to 70
